File: apimodel/src/main/java/com/pgssoft/mvvm/model/api/ApiTable.java
package com.pgssoft.mvvm.model.api;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by bstokrocki on 30.01.2017.
 */
public class ApiTable {
    @SerializedName("table")
    private String id;

    @SerializedName("no")
    private String number;

    @SerializedName("effectiveDate")
    private String lastUpdate;

    @SerializedName("rates")
    private List<ApiRate> apiRates;

    public ApiTable(String name, String number, String lastUpdate, List<ApiRate> apiRates) {
        this.id = name;
        this.number = number;
        this.lastUpdate = lastUpdate;
        this.apiRates = apiRates;
    }

    public String getId() {
        return id;
    }

    public String getNumber() {
        return number;
    }

    public String getLastUpdate() {
        return lastUpdate;
    }

    public List<ApiRate> getApiRates() {
        return apiRates;
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/model/api/ApiTable.kt
package com.pgssoft.kotlinplayground.model.api

import com.google.gson.annotations.SerializedName

/**
 * Created by bstokrocki on 30.01.2017.
 */
class ApiTable(@SerializedName("table") val id: String,
               @SerializedName("no") val number: String,
               @SerializedName("effectiveDate") val lastUpdate: String,
               @SerializedName("rates") val apiRates: List<ApiRate>)
